File: sast-python2/case/completeness/single_app_tracing/alias/alias_001_T.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = false
# evaluation item = 完整度->单应用跟踪完整度->别名
# scene introduction = 别名问题
# level = 2
# bind_url = completeness/single_app_tracing/alias/alias_001_T
# evaluation information end
import os


def alias_001_T(taint_src):
    a = {u'value': u'_'}
    b = a
    b[u'value'] = taint_src
    taint_sink(a[u'value'])


def taint_sink(o):
    os.system(o)


# 示例调用
if __name__ == u"__main__":
    taint_src = u"taint_src_value"
    alias_001_T(taint_src)

File: sast-php/case/completeness/single_app_tracing/alias/alias_001_T.php
<?php
// evaluation information start
// real case = true
// evaluation item = 完整度->单应用追踪->别名
// scene introduction = 普通变量赋值传递污点
// level = 2
// bind_url = completeness/single_app_tracing/alias/alias_001_T
// evaluation information end

function alias_001_T($__taint_src) {
    $a = $__taint_src;
    $b = $a;
    __taint_sink($b);
}

function __taint_sink($o) {
    shell_exec($o);
}

$taint_src = "taint_src_value";
alias_001_T($taint_src);
